crates/telio-lana/src/lib.rs

```diff
@@ -115,3 +135,196 @@ pub fn deinit_lana() -> Result<moose::Result, moose::Error> {
+                        LANA_APP_NAME.to_string(),
+                        app_version.clone(),
+                        LANA_MOOSE_VERSION.to_string(),
+                        prod,
+                        Box::new(moose_callbacks::MooseInitCallback {
+                            init_success_tx: tx,
+                        }),
+                        Box::new(moose_callbacks::MooseErrorCallback),
+                    )
+                } else {
+                    moose::init(
+                        event_path,
+                        LANA_APP_NAME.to_string(),
+                        app_version,
+                        LANA_MOOSE_VERSION.to_string(),
+                        prod,
+                        Box::new(moose_callbacks::MooseInitCallback {
+                            init_success_tx: tx,
+                        }),
+                        Box::new(moose_callbacks::MooseErrorCallback),
+                    )
+                }
+            })
+        } else {
+            moose::init(
+                event_path,
+                LANA_APP_NAME.to_string(),
+                app_version,
+                LANA_MOOSE_VERSION.to_string(),
+                prod,
+                Box::new(moose_callbacks::MooseInitCallback {
+                    init_success_tx: tx,
+                }),
+                Box::new(moose_callbacks::MooseErrorCallback),
+            )
+        }
+    }
+}
+#[cfg(test)]
+mod test {
+    use super::event_log::moose;
+    use super::moose::{ErrorCallback, InitCallback};
+    use super::RefCell;
+    use super::{deinit_lana, init_lana, is_lana_initialized};
+
+    thread_local!(pub static MOCK: RefCell<Option<MooseMock>> = RefCell::new(None));
+
+    pub struct MooseMock {
+        result_fn: bool,
+        result_cb: bool,
+        timeout_cb: bool,
+    }
+
+    impl MooseMock {
+        #[allow(clippy::too_many_arguments)]
+        pub fn init(
+            &self,
+            _event_path: String,
+            _app_name: String,
+            _app_version: String,
+            _exp_moose_ver: String,
+            _prod: bool,
+            init_cb: Box<(dyn InitCallback + 'static)>,
+            _error_cb: Box<(dyn ErrorCallback + Sync + std::marker::Send + 'static)>,
+        ) -> std::result::Result<moose::Result, moose::Error> {
+            if !self.timeout_cb {
+                if self.result_cb {
+                    init_cb.after_init(&Ok(moose::TrackerState::Ready));
+                } else {
+                    init_cb.after_init(&Err(moose::MooseError::NotInitiated));
+                }
+            }
+
+            if self.result_fn {
+                Ok(moose::Result::Success)
+            } else {
+                Err(moose::Error::NotInitiatedError)
+            }
+        }
+    }
+
+    impl Default for MooseMock {
+        fn default() -> Self {
+            MooseMock {
+                result_fn: true,
+                result_cb: true,
+                timeout_cb: false,
```

Comment:
I think that both `_cb` and `_fn` suffixes are usually used for fields that store something callable, so some better names would be nice here. I would go with (in order): `return_success`, `init_cb_status` and `should_call_init_cb`. Additionally I would change the type of `init_cb_status` to be `Result<TrackerState, MooseError>`.

Reply:
I agree, names changed.